src/main/java/io/vertx/ext/mail/mailencoder/Utils.java

```diff
@@ -92,12 +92,27 @@ static boolean mustEncode(String s) {
 
   private static final AtomicInteger count = new AtomicInteger(0);
 
-  static String generateBoundary() {
-    return "=--vertx_mail_" + Thread.currentThread().hashCode() + "_" + System.currentTimeMillis() + "_" + count.getAndIncrement();
+  private static String nomaliseUserAgent(String userAgent) {
+    return userAgent.replace(" ", "_");
   }
 
-  static String generateMessageID(String hostname) {
-    return "<msg." + System.currentTimeMillis() + ".vertxmail." + count.getAndIncrement() + "@" + hostname + ">";
+  static String generateBoundary(String userAgent) {
+    return new StringBuilder().append("=--")
+      .append(nomaliseUserAgent(userAgent)).append("_")
+      .append(Thread.currentThread().hashCode()).append("_")
+      .append(System.currentTimeMillis()).append("_")
+      .append(count.getAndIncrement())
+      .toString();
+  }
```

Comment:
I believe using StringBuilder and toString() is not needed when creating a string in a single statement as this is done internally in an optimized way, please use a normal string expression to keep it simple

Reply:
Thanks for the review, I have updated as required, please review again, thanks :)